#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 33
  FPS 35
  optimalThreadCount 4
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.08800974826213012 -0.9438255021319816 -0.3185085646197751 3.67017485990578
  position 1.0451463751726369 1.8303504373077495 -1.9607502142970195
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  wallHeight 0.4
}
Robot {
  translation 0.08789162143837381 0.03971225991353594 -0.22944149416640322
  rotation -3.8920177435596236e-08 0.9999999999999992 5.222918254003049e-12 -3.1413189314001877
  children [
    DistanceSensor {
      name "path-following-IR-sensor-right"
    }
    DistanceSensor {
      name "path-following-IR-sensor-left"
    }
    Compass {
    }
    GPS {
      translation 0 1 0
    }
    RangeFinder {
      translation 0 1 0
      rotation 0 1 0 -3.141592653589793
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.05 0.01
          }
        }
      ]
      fieldOfView 1.3
      width 640
      height 480
      maxRange 10
    }
    Transform {
      translation 0 1 0
      rotation 0 1 0 3.1415926535
      children [
        Camera {
          rotation 1 0 0 0
          fieldOfView 1.3
          width 640
          height 480
          near 0.002
          far 100
        }
      ]
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 735.725478176064
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation -1 0 0 5.690388071127175
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity -1.601741625867028e-13 3.7879205583588304e-08 1.982039601324138e-12
        angularVelocity 5.3192588600859765e-11 9.980004261875275e-12 7.267535677395163e-14
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 716.7333125071194
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 1 0 0 0.4501874886465638
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity -1.5658427856067757e-13 3.787749097306692e-08 -1.4611453525824569e-12
        angularVelocity -5.813865835865305e-11 9.979994970428456e-12 1.1802119085299354e-13
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 735.7255125448362
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation -1 0 0 5.690353702355012
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 1.5026948209551453e-13 3.787922781433563e-08 1.981825711082522e-12
        angularVelocity 5.319205882177778e-11 9.980004485678636e-12 4.11372801785766e-14
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 716.7333461748984
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 0.9999999999999999 0 0 0.4502211564255703
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 1.550148641091178e-13 3.787751320381612e-08 1.734866513707941e-12
        angularVelocity 5.770556409683397e-11 9.980004189472684e-12 5.008146568165719e-14
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  name "robot(1)"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "robot_controller"
}
Desk {
  translation 6.96416 0 5.236
  rotation 0 1 0 -1.57
}
Desk {
  translation -2 4.68 0
  rotation 0 1 0 1.57
  name "desk(1)"
}
Desk {
  translation 5.19 0 2
  rotation 1 0 0 0.785398
  name "desk(2)"
}
Desk {
  translation 0 4.6 -2
  name "desk(3)"
}
Can {
  translation -2.19 0.060958366407094546 -4.42458115371223e-15
  rotation 0.008834291396915028 -0.9998566559978774 -0.014443779015779343 1.7736422317643532e-15
}
Pedestrian {
  translation 1.29 0.44 3.30546e-07
  rotation 0 1 0 -1.5707853071795865
}
Pedestrian {
  translation -1.22 0.44 0.100029
  rotation 0 1 0 1.57081
  name "pedestrian(1)"
}
Pedestrian {
  translation -0.069983 0.44 1.19002
  rotation 0 1 0 -3.141575307179586
  name "pedestrian(2)"
}
Pedestrian {
  translation 0.0299722 0.44 -1.29998
  rotation 0 1 0 1.73368e-05
  name "pedestrian(3)"
}
